database/guests.ts

```diff
@@ -1,44 +1,55 @@
-declare module globalThis {
-  let guestsDatabase: Guest[];
-}
-
-// Initialize database to empty array only once
-//
-// TODO: Add a full PostgreSQL database connection
-if (!('guestsDatabase' in globalThis)) {
-  globalThis.guestsDatabase = [];
-}
-
-const guests = globalThis.guestsDatabase;
-
-console.log('guests in db', guests);
-
-export type Guest = {
-  id: number;
-  firstName: string;
-  lastName: string;
-  attending: boolean;
-};
-
-export function getGuests() {
+import { Guest } from '../migrations/00000-createTableGuests';
+import { sql } from './connect';
+
+export const getGuests = async () => {
+  const guests = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+  `;
   return guests;
-}
-
-export function getGuest(id: number) {
-  return guests.find((guest: Guest) => Number(guest.id) === Number(id));
-}
+};
 
-export function addGuest(guest: Guest) {
-  guests.push(guest);
-  return guests;
-}
+export const getGuest = async (id: number) => {
+  const [guest] = await sql<Guest[]>`
+    SELECT
+      *
+    FROM
+      guests
+    WHERE
+      id = ${id}
+  `;
+  return guest;
+};
 
-export function deleteGuest(id: number) {
-  const index = guests.findIndex((guest) => guest.id === id);
+export const addGuest = async (newGuest: Omit<Guest, 'id'>) => {
+  const [guest] = await sql<Guest[]>`
+      INSERT INTO
+        guests (
+          first_name,
+          last_name,
+          attending
+        )
+      VALUES
+        (
+          ${newGuest.firstName},
+          ${newGuest.lastName},
+          ${newGuest.attending}
+        )
+      RETURNING
+        guests.*
+    `;
+  return guest;
+};
 
-  if (index === -1) {
-    return;
-  }
-  guests.splice(index, 1);
-  return guests;
-}
+export const deleteGuest = async (removeGuest: Pick<Guest, 'id'>) => {
+  const [guest] = await sql<Guest[]>`
+      DELETE FROM guests
+      WHERE
+        id = ${removeGuest.id}
+      RETURNING
+        guests.*
+    `;
```

Comment:
indentation is off here (seems like it's off in other queries too) - I guess you don't have the Prettier plugin set up?

Reply:
oh, looks like the ESLint plugin hasn't been set up completely (Prettier deps missing):

https://github.com/upleveled/eslint-config-upleveled/blob/6c24362091950278f1b94bbc5aaf4de7f5aebb04/bin/install.js#L99-L108

Reply:
Yes, forgot to run ESLint the way I showed it in my video. Now Prettier will format
[fb94173](https://github.com/upleveled/expo-example-spring-2024-atvie/pull/4/commits/fb941732d03c775317e4755af9bbb36c33e9d80d)